kafka-connect/kafka-connect-events/src/main/java/org/apache/iceberg/connect/events/Event.java

```diff
@@ -0,0 +1,169 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.connect.events;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.UUID;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.apache.iceberg.avro.AvroEncoderUtil;
+import org.apache.iceberg.avro.AvroSchemaUtil;
+import org.apache.iceberg.data.avro.DecoderResolver;
+
+/**
+ * Class representing all events produced to the control topic. Different event types have different
+ * payloads.
+ */
+public class Event implements Element {
+
+  private UUID id;
+  private EventType type;
+  private Long timestamp;
+  private String groupId;
+  private Payload payload;
+  private final Schema avroSchema;
+
+  public static byte[] encode(Event event) {
+    try {
+      return AvroEncoderUtil.encode(event, event.getSchema());
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static Event decode(byte[] bytes) {
+    try {
+      Event event = AvroEncoderUtil.decode(bytes);
+      // clear the cache to avoid memory leak
+      DecoderResolver.clearCache();
+      return event;
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  // Used by Avro reflection to instantiate this class when reading events
+  public Event(Schema avroSchema) {
+    this.avroSchema = avroSchema;
+  }
+
+  public Event(String groupId, EventType type, Payload payload) {
+    this.id = UUID.randomUUID();
+    this.type = type;
+    this.timestamp = System.currentTimeMillis();
+    this.groupId = groupId;
+    this.payload = payload;
+
+    this.avroSchema =
+        SchemaBuilder.builder()
+            .record(getClass().getName())
+            .fields()
+            .name("id")
+            .prop(AvroSchemaUtil.FIELD_ID_PROP, 1500)
+            .type(UUID_SCHEMA)
+            .noDefault()
+            .name("type")
+            .prop(AvroSchemaUtil.FIELD_ID_PROP, 1501)
+            .type()
+            .intType()
+            .noDefault()
+            .name("timestamp")
+            .prop(AvroSchemaUtil.FIELD_ID_PROP, 1502)
+            .type()
+            .longType()
+            .noDefault()
+            .name("payload")
+            .prop(AvroSchemaUtil.FIELD_ID_PROP, 1503)
+            .type(payload.getSchema())
+            .noDefault()
+            .name("groupId")
+            .prop(AvroSchemaUtil.FIELD_ID_PROP, 1504)
+            .type()
+            .stringType()
+            .noDefault()
+            .endRecord();
+  }
+
+  public UUID id() {
+    return id;
+  }
+
+  public EventType type() {
+    return type;
+  }
+
+  public Long timestamp() {
+    return timestamp;
+  }
+
+  public Payload payload() {
+    return payload;
+  }
+
+  public String groupId() {
+    return groupId;
+  }
+
+  @Override
+  public Schema getSchema() {
```

Comment:
minor: I believe we typically don't use a `get` prefix in the Iceberg codebase

Reply:
I left this one in as it is an override